sycl/include/CL/sycl/image.hpp

```
@@ -105,38 +194,68 @@ class image {
 
+  // Returns the total number of elements in the image
   size_t get_count() const { return impl->get_count(); }
 
+  // Returns the allocator provided to the image
   AllocatorT get_allocator() const { return impl->get_allocator(); }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::image>
-  get_access(handler &commandGroupHandler) {
-    return impl->template get_access<dataT, accessMode>();
+  //// Provide an accessor to the image with target type image.
+  // template <typename dataT, access::mode accessMode, typename = typename
+  // std::enable_if<detail::is_validImageDataT<dataT>::value,dataT>::type>
+  // accessor<dataT, dimensions, accessMode, access::target::image>
+  // get_access(handler &CommandGroupHandler) {
+  // assert(!"Not Implemented");
+  // return;
+  // // TODO: Define the accessor constructors for images and implement the
+  // APIs.
+  // // return accessor<dataT, dimensions, accessMode, access::target::image,
+  // // access::placeholder::false_t>(*this, CommandGroupHandler);
+  //}
+
+  // template <typename dataT, access::mode accessMode>
+  // accessor<dataT, dimensions, accessMode, access::target::host_image>
+  // get_access() {
+  // assert(!"Not Implemented");
+  // return;
+  // // TODO: Define the accessor constructors for images and implement the
+  // APIs.
+  // // return accessor<dataT, dimensions, accessMode,
+  // access::target::host_image,
+  // // access::placeholder::false_t>(*this);
+  //}
+
+  template <typename Destination = std::nullptr_t>
+  void set_final_data(Destination FinalData = nullptr) {
+    assert(!"Not Implemented");
+    return;
   }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::host_image>
-  get_access() {
-    return impl->template get_access<dataT, accessMode>();
+  void set_write_back(bool Flag = true) {
+    assert(!"Not Implemented");
```

I think it's better to throw an exception here.

---

Changed.